### PR TITLE
Nr 54707-Nr54704  cockroach db quick start alerts and dashboard

### PR DESCRIPTION
**Pull request by gsidhwani-nr**
# Summary

A concise description of the changes being introduced. Please review the pre-merge checklist section to validate this pull request is ready for review and merge. **If it is not ready, please mark the pull request as a draft.**

## Pre merge checklist

- [ X] Did you check you NRQL syntax? - Does it work?
- [x] Did you include an InstallPlan and Documentation reference?
- [x] Are all documentation links publically accessible?
- [X] Did you check your descriptive content for [voice and tone](https://docs.newrelic.com/docs/style-guide/writing-strategies/voice-strategies-docs-sound-new-relic/)? 
- [X ] Did you check your descriptive content for spelling and grammar errors?

### Dashboards 

- [x] Does the PR contain a screenshot for each of your dashboards?
- [x] Do your screenshots show data?
- [x] Has the [sanitization script](https://github.com/newrelic/newrelic-quickstarts/blob/main/CONTRIBUTING.md#dashboards) been run on each dashboard?

### Alerts

- [X] Did you check that your alerts actually work?

https://staging.onenr.io/0gbRK6OBOwE

### CODE REVIEW

File: alert-policies/cockroach-db/Low_disk_storage.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+name: Low disk storage 
+
+description: |+
+  Send an alert when storage capacity is available below 15% .The alert threshold can be changed to your desired storage limit requirements.
```
Comment (thezackm) [inline alert-policies/cockroach-db/Low_disk_storage.yml:4]: please update for readability: 

```
Send an alert when available storage capacity is below 15% for over 15 minutes. The alert threshold can be changed to your desired storage limit requirements.
```

File: alert-policies/cockroach-db/ca_certificate expiration.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
+name: CA certificate expiration 
+
+description: |+
+  This alert is triggered when there are less than 90 days remaining for certificate expiration. The alert threshold can be changed to your desired expiration alerting requirements.
```
Comment (thezackm) [inline alert-policies/cockroach-db/ca_certificate expiration.yml:4]: please update for readability: 

```
This alert is triggered when there are less than 90 days remaining for CA certificate expiration. The alert threshold can be changed to your desired expiration alerting requirements.
```

File: alert-policies/cockroach-db/client_ca_certificate expiration.yml [hunk 3]
```diff
@@ -0,0 +1,25 @@
+name: Client CA certificate expiration 
+
+description: |+
+  This alert is triggered when there are less than 90 days remaining for certificate expiration. The alert threshold can be changed to your desired expiration alerting requirements.
```
Comment (thezackm) [inline alert-policies/cockroach-db/client_ca_certificate expiration.yml:4]: please update for readability: 

```
This alert is triggered when there are less than 90 days remaining for Client CA certificate expiration. The alert threshold can be changed to your desired expiration alerting requirements.
```

File: alert-policies/cockroach-db/node_certificate expiration.yml [hunk 4]
```diff
@@ -0,0 +1,26 @@
+name: Node certificate expiration 
+
+description: |+
+  This alert is triggered when there are less than 30 days remaining for certificate expiration. The alert threshold can be changed to your desired expiration alerting requirements.
```
Comment (thezackm) [inline alert-policies/cockroach-db/node_certificate expiration.yml:4]: please update for readability: 

```
This alert is triggered when there are less than 30 days remaining for Node certificate expiration. The alert threshold can be changed to your desired expiration alerting requirements.
```

File: alert-policies/cockroach-db/ui_ca_certificate expiration.yml [hunk 5]
```diff
@@ -0,0 +1,26 @@
+name: UI CA certificate expiration 
+
+description: |+
+  This alert is triggered when there are less than 90 days remaining for certificate expiration. The alert threshold can be changed to your desired expiration alerting requirements.
```
Comment (thezackm) [inline alert-policies/cockroach-db/ui_ca_certificate expiration.yml:4]: please update for readability: 

```
This alert is triggered when there are less than 90 days remaining for UI CA certificate expiration. The alert threshold can be changed to your desired expiration alerting requirements.
```

File: alert-policies/cockroach-db/ui_certificate expiration.yml [hunk 6]
```diff
@@ -0,0 +1,26 @@
+name: UI certificate expiration 
+
+description: |+
+  This alert is triggered when there are less than 30 days remaining for certificate expiration. The alert threshold can be changed to your desired expiration alerting requirements.
```
Comment (thezackm) [inline alert-policies/cockroach-db/ui_certificate expiration.yml:4]: please update for readability: 

```
This alert is triggered when there are less than 30 days remaining for UI certificate expiration. The alert threshold can be changed to your desired expiration alerting requirements.
```

File: alert-policies/cockroach-db/cluster_running_with_different_versions.yml [hunk 7]
```diff
@@ -0,0 +1,26 @@
+name: Cluster running with different versions 
+
+description: |+
+  Send an alert Cluster running with different versions of cockroachDB .
```
Comment (thezackm) [inline alert-policies/cockroach-db/cluster_running_with_different_versions.yml:4]: please update for readability: 

```
This alert is triggered when there are multiple versions of CockroachDB running in a single cluster.
```

File: alert-policies/cockroach-db/high_fd_open_count.yml [hunk 8]
```diff
@@ -0,0 +1,26 @@
+name: High fd open count 
+
+description: |+
+  Send an alert when a node is getting 80% of the open file descriptor limit.The alert threshold can be changed to your desired fd limit requirements.
```
Comment (thezackm) [inline alert-policies/cockroach-db/high_fd_open_count.yml:4]: please update for readability: 

```
This alert is triggered when a node passes 80% of the open file descriptors limit for 15 minutes. The alert threshold can be changes to your desired file descriptors limit requirements.
```

File: alert-policies/cockroach-db/instance_dead.yml [hunk 9]
```diff
@@ -0,0 +1,25 @@
+name: Instance dead alert  
+
+description: |+
+  Send an alert when an instance is down for 15 minutes. The alert threshold can be changed to your acceptable downtime requirements.
```
Comment (thezackm) [inline alert-policies/cockroach-db/instance_dead.yml:4]: please update for readability: 

```
This alert triggers when an instance is down for 15 minutes. The alert threshold can be changed to your acceptable downtime requirements.
```

File: alert-policies/cockroach-db/node_down.yml [hunk 10]
```diff
@@ -0,0 +1,26 @@
+name: node down alert
```
Comment (thezackm) [inline alert-policies/cockroach-db/node_down.yml:1]: change to `Node down alert` (sentence case)

File: alert-policies/cockroach-db/node_down.yml [hunk 11]
```diff
@@ -0,0 +1,26 @@
+name: node down alert 
+
+description: |+
+  Send an alert when an node is down for 15 minutes. The alert threshold can be changed to your acceptable downtime requirements.
```
Comment (thezackm) [inline alert-policies/cockroach-db/node_down.yml:4]: please update for readability: 

```
This alert triggers when a node is down for 15 minutes. The alert threshold can be changed to your acceptable downtime requirements.
```

File: alert-policies/cockroach-db/slow_latch_requests.yml [hunk 12]
```diff
@@ -0,0 +1,29 @@
+name: Slow lease requests
```
Comment (thezackm) [inline alert-policies/cockroach-db/slow_latch_requests.yml:1]: change to `Slow latch requests` (to match correct metric in the alert)

File: alert-policies/cockroach-db/slow_latch_requests.yml [hunk 13]
```diff
@@ -0,0 +1,29 @@
...
+WHERE (app_kubernetes_io_name LIKE '%cockroach%' or app_newrelic_com_name LIKE '%cockroach%') AND (instrumentation.name = 'remote-write') "
+
+# Function used to aggregate the NRQL query value(s) for comparison to the terms.threshold (Default: SINGLE_VALUE)
+valueFunction: SINGLE_VALUE
+
+# List of Critical and Warning thresholds for the condition
+terms:
+  - priority: CRITICAL
+    # Operator used to compare against the threshold.
+    operator: EQUALS
```
Comment (thezackm) [inline alert-policies/cockroach-db/slow_latch_requests.yml:19]: this should be `ABOVE`

File: alert-policies/cockroach-db/slow_latch_requests.yml [hunk 14]
```diff
@@ -0,0 +1,29 @@
...
+    # Value that triggers a violation
+    threshold: 1
```
Comment (thezackm) [inline alert-policies/cockroach-db/slow_latch_requests.yml:21]: this should be `0`

(alert me when > 0)

File: alert-policies/cockroach-db/slow_lease_requests.yml [hunk 15]
```diff
@@ -0,0 +1,29 @@
+name: Slow lease requests 
+
+description: |+
+  Slow lease requests
```
Comment (thezackm) [inline alert-policies/cockroach-db/slow_lease_requests.yml:4]: please update for readability: 

```
This alert triggers when there are > 0 slow lease requests for over 5 minutes. The alert threshold can be changed to your acceptable slow request requirements.
```

File: alert-policies/cockroach-db/slow_lease_requests.yml [hunk 16]
```diff
@@ -0,0 +1,29 @@
...
+WHERE (app_kubernetes_io_name LIKE '%cockroach%' or app_newrelic_com_name LIKE '%cockroach%') AND (instrumentation.name = 'remote-write') "
+
+# Function used to aggregate the NRQL query value(s) for comparison to the terms.threshold (Default: SINGLE_VALUE)
+valueFunction: SINGLE_VALUE
+
+# List of Critical and Warning thresholds for the condition
+terms:
+  - priority: CRITICAL
+    # Operator used to compare against the threshold.
+    operator: EQUALS
```
Comment (thezackm) [inline alert-policies/cockroach-db/slow_lease_requests.yml:19]: this should be `ABOVE`

File: alert-policies/cockroach-db/slow_lease_requests.yml [hunk 17]
```diff
@@ -0,0 +1,29 @@
...
+    # Value that triggers a violation
+    threshold: 1
```
Comment (thezackm) [inline alert-policies/cockroach-db/slow_lease_requests.yml:21]: this should be `0`

File: alert-policies/cockroach-db/slow_raft_requests.yml [hunk 18]
```diff
@@ -0,0 +1,29 @@
+name: Slow raft requests 
+
+description: |+
+  Slow raft requests
```
Comment (thezackm) [inline alert-policies/cockroach-db/slow_raft_requests.yml:4]: please update for readability: 

```
This alert triggers when there are > 0 slow raft requests for over 5 minutes. The alert threshold can be changed to your acceptable slow request requirements.
```

File: alert-policies/cockroach-db/slow_raft_requests.yml [hunk 19]
```diff
@@ -0,0 +1,29 @@
...
+WHERE (app_kubernetes_io_name LIKE '%cockroach%' or app_newrelic_com_name LIKE '%cockroach%') AND (instrumentation.name = 'remote-write') "
+
+# Function used to aggregate the NRQL query value(s) for comparison to the terms.threshold (Default: SINGLE_VALUE)
+valueFunction: SINGLE_VALUE
+
+# List of Critical and Warning thresholds for the condition
+terms:
+  - priority: CRITICAL
+    # Operator used to compare against the threshold.
+    operator: EQUALS
```
Comment (thezackm) [inline alert-policies/cockroach-db/slow_raft_requests.yml:19]: this should be `ABOVE`

File: alert-policies/cockroach-db/slow_raft_requests.yml [hunk 20]
```diff
@@ -0,0 +1,29 @@
...
+    # Value that triggers a violation
+    threshold: 1
```
Comment (thezackm) [inline alert-policies/cockroach-db/slow_raft_requests.yml:21]: this should be `0`

**Review by thezackm — CHANGES_REQUESTED** [review]
@gsidhwani-nr - review submitted for `prometheus-growth` project.

this PR needs to be combined with PR #1491 and you need to add a `/quickstarts/cockroach-db/config.yml` file and associated logo file that references both the alerts and dashboard.

**Review by thezackm — DISMISSED** [review]
approved for `prometheus-growth` project; thanks!

**Review by zstix — APPROVED** [review]

File: quickstarts/cockroach-db/config.yml [hunk 21]
```diff
@@ -0,0 +1,48 @@
+id: e2ee8b6e-5c2a-11ed-9b6a-0242ac120002
+slug: cockroach database-newrelic
```
Comment (zstix) [inline quickstarts/cockroach-db/config.yml:2]: The slug can only have letters, numbers, and hyphens. Perhaps we should use this instead:

```suggestion
slug: cockroach-database-newrelic
```

**Review by zstix — CHANGES_REQUESTED** [review]

**Comment by gsidhwani-nr** [comment]
changes done. cockroach-database-newrelic

**Review by zstix — APPROVED** [review]

**Pull request merged**